### PR TITLE
[nmcli] add runner and runner-hwaddr-policy for network teaming

### LINKED ISSUE #1950: The mode parameter of community.general.nmcli does not take effect for team connection
### Summary

I want to create a connection of type team, mode is active-backup. But the setting does not take effect

### Issue Type

Bug Report

### Component Name

nmcli

### Ansible Version

```
ansible 2.10.6
python version = 3.7.6 (default, Dec 30 2019, 19:38:28) [Clang 11.0.0 (clang-1100.0.33.16)]

```

### Configuration

```paste below

```

### OS / Environment

_No response_

### Steps To Reproduce

```
- name: add team1
      community.general.nmcli:
        type: team
        conn_name: team1
        ip4: 1.1.1.1/24
        gw4: 1.1.1.254
        mode: active-backup
        state: present
```

### Expected Results

_No response_

### Actual Results

```
#nmcli con show team1


team.runner:                            roundrobin


```

### PR DESCRIPTION
**Pull request by omula**
##### SUMMARY

This PR allows to select the runner for a team configuration. It also adds the possibility of select the hwaddr-policy for the activebackup runner.

Fixes #1950

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

nmcli

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/net_tools/nmcli.py [hunk 1]
```diff
@@ -265,6 +265,20 @@
               frame was received on.
         type: bool
         default: yes
+    runner:
+        description:
+            - This is the type of device or network connection that you wish to create for a team.
+        type: str
+        choices: [ broadcast, roundrobin, activebackup, loadbalance, lacp ]
+        default: roundrobin
+        version_added: 3.3.0
```
Comment (russoz) [inline plugins/modules/net_tools/nmcli.py:274]:
```suggestion
        version_added: 3.4.0
```

File: plugins/modules/net_tools/nmcli.py [hunk 2]
```diff
@@ -265,6 +265,20 @@
...
+    runner_hwaddr_policy:
+        description:
+            - This  defines  the policy of how hardware addresses of team device and port devices
+              should be set during the team lifetime.
+        type: str
+        choices: [ same_all, by_active, only_active ]
+        version_added: 3.3.0
```
Comment (russoz) [inline plugins/modules/net_tools/nmcli.py:281]:
```suggestion
        version_added: 3.4.0
```

**Review by russoz — APPROVED** [review]
Updating the `version_added`, as 3.3.0 has already been released

**Comment by russoz** [comment]
Hi @omula , slightly on the margin of this PR, but would you mind taking a look if it addresses (or if it could be easily adjusted to address) #473 as well?

**Comment by felixfontein** [comment]
This PR is a feature PR, not a bugfix PR btw, since it adds new features.

In any case, could you please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#changelogs-how-to)?

**Comment by omula** [comment]
@felixfontein I marked it as bugfix because in the documentation it is said to work with the mode parameter while it wasn't doing anything. But you are also right in the sense that instead of using mode I created a new parameter, to match team naming and I added a new one. Do you want me to change the description.

Going to add the fragment. Thanks for reviewing it.

**Comment by aminvakil** [comment]
> Do you want me to change the description.

Yes, please. Although this does not have any effect on the merging commit, it will prevent confusion later someone reading PR.

File: plugins/modules/net_tools/nmcli.py [hunk 3]
```diff
@@ -265,6 +265,20 @@
...
+    runner:
+        description:
+            - This is the type of device or network connection that you wish to create for a team.
+        type: str
+        choices: [ broadcast, roundrobin, activebackup, loadbalance, lacp ]
+        default: roundrobin
+        version_added: 3.4.0
+    runner_hwaddr_policy:
+        description:
+            - This  defines  the policy of how hardware addresses of team device and port devices
```
Comment (aminvakil) [inline plugins/modules/net_tools/nmcli.py:277]:
```suggestion
            - This defines the policy of how hardware addresses of team device and port devices
```

**Review by aminvakil — APPROVED** [review]
Not a `nmcli` expert, but code and tests LGTM.

**Review by russoz — APPROVED** [review]
Not very familiar with nmcli, but it looks good to me.

**Comment by felixfontein** [comment]
Me neither, but looks good as well :)

**Comment by felixfontein** [comment]
@omula thanks for contributing this!
@russoz @aminvakil thanks for reviewing!

**Pull request merged**